x-pack/plugins/lens/public/types.ts

```diff
@@ -224,6 +224,10 @@ export interface Datasource<T = unknown, P = unknown> {
    * uniqueLabels of dimensions exposed for aria-labels of dragged dimensions
    */
   uniqueLabels: (state: T) => Record<string, string>;
+  /**
+   * Check the internal state integrity and returns a list of missing references
```

Comment:
nit: Please make this include the word "index pattern" - I know it's obvious right now, but it's not for the data sources we plan to work with

Reply:
My idea is to not mention index pattern here, rather be generic. The fact that we have only index pattern right now should not be of interest from outside. Perhaps it would return an array of `{id: string, type: 'indexpattern'}` instead of simple ids, to make aware the consumer where to look for those ids.